backend/tarsy/controllers/history_controller.py

```
@@ -314,4 +315,162 @@ async def get_filter_options(
+            f"Session {session_id} status changed to {session.status} just before timeout"
+        )
+
+
+@router.post(
+    "/sessions/{session_id}/cancel",
+    summary="Cancel Session",
+    description="Cancel an active alert processing session"
+)
+async def cancel_session(
+    session_id: str = Path(..., description="Session ID to cancel"),
+    background_tasks: BackgroundTasks = BackgroundTasks(),
```

Ruff (0.14.1)

Do not perform function call `BackgroundTasks` in argument defaults; instead, perform the call within the function, or read the default from a module-level singleton variable

(B008)